server/services/nl.py

```diff
@@ -120,11 +234,38 @@
 }
 
 
+def _add_to_set_from_list(set_strings, list_string):
```

Comment:
would be nice to start adding tests for these util functions, then we would expand as we see issues from demo.

Reply:
Done for some of the functions introduced in this PR